### PR TITLE
Adding an error code type and network request method to support repre…

### PR DESCRIPTION
**Pull request by jdizoglio**
…sentation changes.

Changes are made at the request of the representations [PR(512)](https://github.com/box/box-ios-preview-sdk-dev/pull/512) for the Preview SDK.

### CODE REVIEW

**Comment by boxcla** [comment]
Verified that @jdizoglio has signed the CLA. Thanks for the pull request!

**Comment by jdizoglio** [comment]
test this please

File: BoxContentSDK/BoxContentSDK/Models/BOXFile.m [hunk 1]
```diff
@@ -3,11 +3,13 @@
 //  BoxContentSDK
 //
 //
-
+ 
 #import "BOXFile.h"
 #import "BOXFileLock.h"
 #import "BOXRepresentation.h"
 
+//#define BOX_REPRESENTATIONS_ENDPOINT
```
Comment (boxtcarpel) [inline BoxContentSDK/BoxContentSDK/Models/BOXFile.m]: keep this?

File: BoxContentSDK/BoxContentSDK/Requests/BOXRequest.m [hunk 2]
```diff
@@ -17,6 +17,7 @@
 
 #define BOX_API_MULTIPART_FILENAME_DEFAULT (@"upload")
 
+//#define BOX_REPRESENTATIONS_ENDPOINT
```
Comment (boxtcarpel) [inline BoxContentSDK/BoxContentSDK/Requests/BOXRequest.m]: keep?

**Comment by boxtcarpel** [comment]
I think we want to revert the file and request model objects, otherwise LGTM

+2

🍔

File: BoxContentSDK/BoxContentSDK/Requests/BOXRequest.m (hunk 2 above)
Comment (jdizoglio) [reply BoxContentSDK/BoxContentSDK/Requests/BOXRequest.m]: done

File: BoxContentSDK/BoxContentSDK/Models/BOXFile.m (hunk 1 above)
Comment (jdizoglio) [reply BoxContentSDK/BoxContentSDK/Models/BOXFile.m]: done

**Pull request merged**